_posts/pl/2022-03-16-npx.md

```diff
@@ -0,0 +1,37 @@
+---
+layout:    post
+title:     "Czy wiesz, czym jest npx?"
+date:      2022-03-18 8:00:00 +0100
+published: true
+didyouknow: true
+lang: pl
+author:    mchrapkowicz
+image:     /assets/img/posts/2022-03-16-npx/keyboard.jpg
+tags:
+- npm
+- npx
+- node
+---
+
+## Czym jest npx?
+
+npx to narzędzie CLI (Command Line Interface) pozwalające wykonywać skrypty pochodzące z repozytorium npm bez zbędnej konfiguracji. Co ciekawe - można w ten sposób wywołać polecenie z pakietu niezainstalowanego na maszynie lokalnej.
+
+Dla przykładu:
+```
+npx create-react-app moja-aplikacja
+```
+
+Powyższe polecenie zainstaluje tymczasowo pakiet 'create-react-app' na maszynie i go wywoła. Po zakończeniu wykonywania w systemie nie będzie żadnej informacji o pakiecie, co można potwierdzić wykonując chociażby polecenie which create-react-app.
+
+## Obsługa pakietów globalnych
+
+Pakiety npm bardzo często są tworzone jako globalne, żeby zaraz po instalacji mogły być wykonywane. To podejście powoduje, że nie można mieć więcej, niż jednej wersji tego samego globalnego pakietu w systemie. npx adresuje ten problem, gdyż potrafi wykonywać skrypty zainstalowane w node_modules konkretnego projektu.
+
+## Instalacja
+
+npx jest wbudowany w npm od wersji 5.2.0. Istnieje również możliwość instalacji samego npx poprzez poniższe polecenie:
```

Comment:
NPX/ NPM

Reply:
Nazwy npm i npx są wszędzie pisane małą literą, myślę że warto tak to tutaj w innych miejscach zostawić :)

Reply:
Rozumiem, ale ja akurat pod kątem wizualnym (capslock wyróżnia się z tekstu i bardziej wpada w oko ;) i ja jestem przyzwyczajona do stosowania dużych liter w akronimach. Oczywiście Wasza decyzja :)

Reply:
Jasna sprawa, tak się utarło z nimi po prostu :D